src/main/java/service/impl/BookServiceImpl.java

```diff
@@ -1,28 +1,39 @@
 package service.impl;
 
+import dto.BookDto;
+import dto.CreateBookRequestDto;
+import exception.EntityNotFoundException;
 import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mapper.BookMapper;
 import model.Book;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import repository.BookRepository;
 import service.BookService;
 
+@RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto).toList();
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(
+                "Can't find book by id " + id)
+        );
+        return bookMapper.toDto(book);
```

Comment:
Use capabilities of Optional to reduce code. 
```suggestion
        return bookRepository.findById(id)
        .map(bookMapper::toDto)
        .orElseThrow(() -> new EntityNotFoundException(
                "Can't find book by id " + id)
        );
```